mmconfig

```diff
@@ -2,416 +2,160 @@
-    for KEY in "${REQUIRED_VARIABLES[@]}" ; do
-        VALUE=$(grep "^${KEY}=" "${MM_CONFIG_FILE}" | cut -d= -f2)
-        printf '\t%-40s %-40s\n' "${KEY}" "${VALUE}"
-    done
-    while true ; do
-        EDITOPTION1="Edit config file in nano"
-        EDITOPTION2="Edit config file in TextMate"
-        _report -q "Edit a variable? "
-        PS3="Selection (enter by number)? "
-        select CONFIG_KEY in "${EDITOPTION1}" "${EDITOPTION2}" "${REQUIRED_VARIABLES[@]}" "Quit" ; do
-            break
-        done
-        [ "${CONFIG_KEY}" = "Quit" ] && { echo Goodbye. ; exit 1 ;};
-        [ "${CONFIG_KEY}" = "${EDITOPTION1}" ] && { nano "${MM_CONFIG_FILE}" ; exit 1 ;};
-        [ "${CONFIG_KEY}" = "${EDITOPTION2}" ] && { mate "${MM_CONFIG_FILE}" ; exit 1 ;};
-        echo -n "Enter the value for ${CONFIG_KEY}: "
-        read -e "CONFIG_VALUE"
-        echo "${CONFIG_KEY} is now set to ${CONFIG_VALUE}"
-        _config_edit "${CONFIG_KEY}" "${CONFIG_VALUE}"
-    done
 
+export MAIN_DIALOG=$(cat << MAIN_FORM
+  <window title="mmconfig configuration" resizable="false" window_position="1">
+    <vbox>
+      <text>
+        <label>Set up mmconfig variables</label>
+      </text>
+      <frame Navigate to and select the directories you have created>
+        <hbox>
+          $(funcDirectory OUTDIR_INGESTFILE)
+          $(funcDirectory OUTDIR_INGESTXDCAM)
+          $(funcDirectory OUTDIR_PAPER)
+          $(funcDirectory AIP_STORAGE)
+          $(funcDirectory PODCASTDELIVER)
+        </hbox>
+        <hbox>
+          $(funcDirectory YOUTUBEDELIVER)
+          $(funcDirectory TMPDIR)
+          $(funcDirectory CUSTOM_LOG_DIR)
+          $(funcDirectory LTO_INDEX_DIR)
+          $(funcDirectory LOCAL_MM_DIR)
+        </hbox>
+      </frame>
+      <frame>
+        <hbox homogeneous="true">
+          $(funcEntry REGEX4PODCAST "Enter the regular expression you would like to use to determine if a file qualifies for podcast:" "80")
+          $(funcEntry REGEX4IA "Enter the regular expression you would like to use to upload to the Internet Archive:" "80")
+        </hbox>
+        <hbox homogeneous="true">
+          $(funcEntry OMNEONIP "Please enter the IP address for the server where you would like assets to be delivered:" "80")
+          $(funcEntry OMNEONPATH "Please enter the file path for the server where you would like assets to be delivered:" "80")
+        </hbox> 
+        <hbox homogeneous="true">
+          $(funcEntry LTO_MD5_FLAGS "Please select the flags you would like md5deep to use (ex: md5deep -rel):" "80")
+          $(funcEntry FILE_MAKER_DB "Please enter the name of the filemaker database:" "80")
+        </hbox>
+        <hbox space-fill="false">
+          <text wrap="false" width-chars="60" xalign="0">
+            <label>Please choose Y or N to test and adjust volume during transcoding:</label>
+          </text>
+          <vbox>
+            <radiobutton>
+              $(funcrdoCreate VOLADJUSTY "Y")
```

Comment:
having the function close a node that was opened outside of that function is ... -_-